hypothesis/client
# Replace local `ConfirmDialog`, etc., with shared `ConfirmModal`

This PR replaces the local `ConfirmDialog` component, used by the `prompts` module, with the `ConfirmModal` component from the `frontend-shared` package and removes the then-unused `Dialog` and `ConfirmDialog` components, their tests and styles.

There are no user-facing changes for the vast majority of users. For users with browsers that no longer support `window.confirm` in third-party iframes, they will now see the consolidated version of the confirm-modal UI.

## To Test

* Check out this branch/`make dev`
* Edit `/src/shared/prompts.js` and comment out the following lines:

```
  const result = window.confirm(message);
  if (result || Date.now() - start >= 10) {
    return result;
  }
```
* On a test page in the local dev server, select some text and click `Annotate`.
* Select "Log Out" from the user menu

You should see:

![image](https://user-images.githubusercontent.com/439947/121727990-a8d21480-caba-11eb-9f71-3dcd54d5736d.png)

(This screenshot shows the keyboard-focus active).

You might also try looking at this with the clean theme (it looks identical) by editing `/dev-server/templates/client-config.js.mustache` and uncommenting these settings:

```
      // theme: 'clean',

      // branding: {
      //  appBackgroundColor: 'white',
      //  ctaBackgroundColor: 'rgba(3, 11, 16, 1)',
      //  ctaTextColor: '#eee',
      //  selectionFontFamily: 'helvetica, arial, sans serif'
      //},
```

Part of https://github.com/hypothesis/frontend-shared/issues/83
